src/Transports/MassTransit.EventHubIntegration/EventHubIntegration/EventHubDataReceiver.cs

```diff
@@ -96,50 +71,19 @@ async Task Handle(ProcessEventArgs eventArgs)
 
             try
             {
-                await _dispatcher.Dispatch(context, context).ConfigureAwait(false);
+                await Dispatch(context, context).ConfigureAwait(false);
             }
             finally
             {
                 context.Dispose();
             }
         }
 
-        Task HandleDeliveryComplete()
-        {
-            if (IsStopping)
-            {
-                LogContext.Debug?.Log("Consumer shutdown completed: {InputAddress}", _context.InputAddress);
-
-                _deliveryComplete.TrySetResult(true);
-            }
-
-            return Task.CompletedTask;
-        }
-
-        protected override Task StopAgent(StopContext context)
-        {
-            LogContext.Debug?.Log("Stopping consumer: {InputAddress}", _context.InputAddress);
-
-            SetCompleted(ActiveAndActualAgentsCompleted(context));
-
-            return Completed;
-        }
-
-        async Task ActiveAndActualAgentsCompleted(StopContext context)
+        protected override async Task ActiveAndActualAgentsCompleted(StopContext context)
         {
             var stopProcessing = _client.StopProcessingAsync();
 
-            if (_dispatcher.ActiveDispatchCount > 0)
-            {
-                try
-                {
-                    await _deliveryComplete.Task.OrCanceled(context.CancellationToken).ConfigureAwait(false);
-                }
-                catch (OperationCanceledException)
-                {
-                    LogContext.Warning?.Log("Stop canceled waiting for message consumers to complete: {InputAddress}", _context.InputAddress);
-                }
-            }
+            await base.ActiveAndActualAgentsCompleted(context).ConfigureAwait(false);
```

Comment:
So we'd need both a StopProcessing and ProcessingStopped virtual callback in the ConsumerAgent class it seems.